### PR TITLE
Use dependency versions compatible with 0.12

### PR DESCRIPTION
**Pull request by paulyoung**

### CODE REVIEW

**Comment by paulyoung** [comment]
I think the only remaining issues here are that `purescript@0.12` isn't published to npm, and after that there might be a problem if there isn't a version of `pulp` that works with 0.12

**Comment by hdgarrood** [comment]
The current version of Pulp should work with 0.12? I know about the `pulp init` thing but that shouldn’t affect pre existing libraries like this one.

**Comment by paulyoung** [comment]
@hdgarrood you're right, it does. That was speculation on my part before I got the CI build passing. 

Thanks for confirming.

**Comment by paulyoung** [comment]
Anything else I can do here to help move this along?

**Comment by garyb** [comment]
No, sorry, I'll get it in soon! I have a branch of my own with changes in too, so wanted to merge the two but have been away / dealing with other libraries so keep forgetting about this.

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,13 @@ language: node_js
 dist: trusty
 sudo: required
 node_js: stable
+env:
+  - PATH=$HOME/purescript:$PATH
 install:
+  - TAG=$(wget -q -O - https://github.com/purescript/purescript/releases/latest --server-response --max-redirect 0 2>&1 | sed -n -e 's/.*Location:.*tag\///p')
+  - wget -O $HOME/purescript.tar.gz https://github.com/purescript/purescript/releases/download/$TAG/linux64.tar.gz
+  - tar -xvf $HOME/purescript.tar.gz -C $HOME/
+  - chmod a+x $HOME/purescript
```
Comment (paulyoung) [inline .travis.yml]: @garyb I could revert these changes if you like

File: package.json [hunk 2]
```diff
@@ -8,7 +8,6 @@
   "devDependencies": {
     "pulp": "^11.0.0",
     "purescript-psa": "^0.5.0",
-    "purescript": "^0.11.1",
```
Comment (paulyoung) [inline package.json:11]: @garyb same here

Comment (garyb) [reply package.json:11]: Yeah, I guess now we have a usable version on npm, may as well 🙂

**Comment by garyb** [comment]
I'll get this in after the compiler version thing is done, my changes can wait for a future release.

**Comment by paulyoung** [comment]
@garyb done. Not sure if #41 does anything differently.

**Comment by thomashoneyman** [comment]
@garyb @paulyoung I don't think my PR does anything differently. It preserves the same functionality but goes through and makes all the necessary `0.12` updates. I'd seen this PR but not that it was cut from the `compiler/0.12` branch, and so I think most likely my work is redundant. I can certainly close that if it is superseded by this.

**Comment by thomashoneyman** [comment]
Well -- I suppose one difference is that CI found an error here! I doubt my implementation is any different, though that one passed.

From CI:

```
Checking 'Submultiplicative euclidean function' law for EuclideanRing
999/1000 test(s) passed.
/home/travis/build/garyb/purescript-quickcheck-laws/output/Effect.Exception/foreign.js:29
    throw e;
    ^
Error: Test 106 (seed 1442996306) failed: 
Test returned false
    at Object.exports.error (/home/travis/build/garyb/purescript-quickcheck-laws/output/Effect.Exception/foreign.js:8:10)
    at /home/travis/build/garyb/purescript-quickcheck-laws/output/Test.QuickCheck/index.js:139:81
    at /home/travis/build/garyb/purescript-quickcheck-laws/output/Data.Foldable/index.js:95:76
    at /home/travis/build/garyb/purescript-quickcheck-laws/output/Data.Foldable/index.js:292:24
    at /home/travis/build/garyb/purescript-quickcheck-laws/output/Data.Foldable/index.js:417:46
    at /home/travis/build/garyb/purescript-quickcheck-laws/output/Data.Function/index.js:19:24
    at __do (/home/travis/build/garyb/purescript-quickcheck-laws/output/Test.QuickCheck/index.js:138:122)
    at __do (/home/travis/build/garyb/purescript-quickcheck-laws/output/Test.QuickCheck/index.js:169:68)
    at __do (/home/travis/build/garyb/purescript-quickcheck-laws/output/Test.QuickCheck.Laws.Data.EuclideanRing/index.js:77:214)
    at __do (/home/travis/build/garyb/purescript-quickcheck-laws/output/Test.Prim.Int/index.js:27:176)
* ERROR: Subcommand terminated with exit code 1
```
